### PR TITLE
Universe: use Relay 13 and the new Rust Compiler

### PR DESCRIPTION
**Pull request by mrtnzlml**
This commit upgrades Relay to version 13 and switches from the old
Relay Compiler to the new Rust one (since both things go together).

Basically, the main change is that now we have only one Relay config
for the whole monorepo and the compiler is being executed for the whole
monorepo as well (while being much faster). Additionally, Relay support
is directly integrated into Flow so in many cases I simply removed
previous Flow types (see `useLazyLoadQuery` and `useFragment`).

There is still ongoing effort to improve the Flow types in Relay so not
everything is finalized. For this reason I decided to use "Compat" types
mode. Similarly, some hooks (`useMutation` and `usePreloadedQuery` for
example) still require explicit types information so I didn't change
these yet. Regardless of that, we are pretty close to use "Final" types.
We just need to wait for the Relay team to finish everything.

Many issues were already resolved but there are still some that need to
be fixed (not blocking this PR):

- https://github.com/facebook/relay/pull/3700
- https://github.com/relayjs/eslint-plugin-relay/issues/131
- https://github.com/prettier/prettier/issues/6102

Important links with additional information:

- https://relay.dev/blog/2021/12/08/introducing-the-new-relay-compiler/
- https://github.com/facebook/relay/releases/tag/v13.0.0
- https://github.com/facebook/relay/releases/tag/v13.0.0-rc.0
- https://github.com/facebook/relay/releases/tag/v13.0.0-rc.1
- https://github.com/facebook/relay/releases/tag/v13.0.0-rc.2

### CODE REVIEW

File: src/abacus-backoffice/src/products/__generated__/ProductsCardsInCategoryQuery.graphql.js [hunk 1]
```diff
@@ -244,6 +223,9 @@ return {
   }
 };
 })();
-// prettier-ignore
-(node: any).hash = '1fb8a914e7bbc4c6535f25b4f592cb12';
-export default node;
+
+if (__DEV__) {
+  (node/*: any*/).hash = "978767bfb60ad8c2fbe1d8fad793c608";
```
Comment (mrtnzlml) [inline src/abacus-backoffice/src/products/__generated__/ProductsCardsInCategoryQuery.graphql.js:228]: TODO: weird hash change

```diff
- 1fb8a914e7bbc4c6535f25b4f592cb12
+ 978767bfb60ad8c2fbe1d8fad793c608
```

File: src/abacus-backoffice/src/products/__generated__/ProductsEditLayoutQuery.graphql.js [hunk 2]
```diff
@@ -416,6 +352,12 @@ return {
   }
 };
 })();
-// prettier-ignore
-(node: any).hash = 'a0bfb0ff14c17bd91dd157d472cab49e';
-export default node;
+
+if (__DEV__) {
+  (node/*: any*/).hash = "65d361f91d79ebe6a215a4a69979a04f";
```
Comment (mrtnzlml) [inline src/abacus-backoffice/src/products/__generated__/ProductsEditLayoutQuery.graphql.js:357]: Suspicious change.

File: src/abacus-kochka/src/shop/__generated__/ProductPageLayoutQuery.graphql.js [hunk 3]
```diff
@@ -232,6 +219,12 @@ return {
   }
 };
 })();
-// prettier-ignore
-(node: any).hash = '860c513cb31bbfc2af9dcf31be1b0b33';
-export default node;
+
+if (__DEV__) {
+  (node/*: any*/).hash = "25a2ea0e1e939f54047f9222193604a7";
```
Comment (mrtnzlml) [inline src/abacus-kochka/src/shop/__generated__/ProductPageLayoutQuery.graphql.js:224]: Suspicious change.

File: src/abacus-kochka/src/shop/__generated__/ShopLayoutContentQuery.graphql.js [hunk 4]
```diff
@@ -238,6 +225,12 @@ return {
   }
 };
 })();
-// prettier-ignore
-(node: any).hash = 'f80231d0ee8d704c6ff74da317ee886a';
-export default node;
+
+if (__DEV__) {
+  (node/*: any*/).hash = "58597fd12ee384ee575ed69c889885a3";
```
Comment (mrtnzlml) [inline src/abacus-kochka/src/shop/__generated__/ShopLayoutContentQuery.graphql.js:230]: Suspicious change.

File: src/example-relay/src/Homepage/locations/__generated__/LocationsPaginatedBidirectional.graphql.js [hunk 5]
```diff
@@ -62,7 +70,7 @@ const node: ReaderFragment = {
     "refetch": {
       "connection": null,
       "fragmentPathInResult": [],
-      "operation": require('./LocationsPaginatedBidirectionalRefetchQuery.graphql.js').default
+      "operation": require('./LocationsPaginatedBidirectionalRefetchQuery.graphql')
```
Comment (mrtnzlml) [inline src/example-relay/src/Homepage/locations/__generated__/LocationsPaginatedBidirectional.graphql.js:73]: Check this if everything is OK

File: src/example-relay/src/Homepage/locations/__generated__/LocationsPaginatedRefetch.graphql.js [hunk 6]
```diff
@@ -69,7 +77,7 @@ return {
         "path": (v0/*: any*/)
       },
       "fragmentPathInResult": [],
-      "operation": require('./LocationsPaginatedRefetchRefetchQuery.graphql.js').default
+      "operation": require('./LocationsPaginatedRefetchRefetchQuery.graphql')
```
Comment (mrtnzlml) [inline src/example-relay/src/Homepage/locations/__generated__/LocationsPaginatedRefetch.graphql.js:80]: Check this

File: src/relay/package.json [hunk 7]
```diff
@@ -18,31 +18,27 @@
...
+    "babel-plugin-relay": "^13.0.0",
     "commander": "^8.3.0",
     "is-ci": "^3.0.1",
-    "react-relay": "^12.0.0",
-    "relay-compiler": "^12.0.0",
-    "relay-config": "^12.0.1",
-    "relay-runtime": "^12.0.0"
+    "react-relay": "^13.0.0",
+    "relay-compiler": "^13.0.0",
+    "relay-config": "0.0.0-main-d92b9e55",
```
Comment (mrtnzlml) [inline src/relay/package.json:33]: I will remove Relay Config together with the `fetch-schema.js` bin removal. This config dependency is useless in Relay 13+

**Review by tbergquist-godaddy — APPROVED** [review]

**Pull request merged**